### PR TITLE
Portfix

### LINKED ISSUE #134: imagegw ignores port number for non-standard registry
Hi there,

When defining a non-standard registry thus:

"server": "f.q.d.n:1234",

the imagegw code does:

        if ':' in server:
            (server, port) = server.split(':')
        if protocol == 'http':
            conn = httplib.HTTPConnection(server)

so it pulls out the server and the port number, but then ignores the port number when making a connection (port is also ignored in the https connection as well from what I see).

I think that's simple to fix by doing this instead:

            conn = httplib.HTTPConnection(server, port)

(and presumably the analagous version for the https variant)

cheers!
Chris

### PR DESCRIPTION
**Pull request by scanon**
Fix #134

### CODE REVIEW

File: imagegw/shifter_imagegw/dockerv2.py [hunk 1]
```diff
@@ -89,21 +89,25 @@ def _setup_http_conn(url, cacert=None):
...
+        port=80
+    else:
+        port = 443
     if url.find('/') >= 0:
         (server, _) = url.split('/', 1)
     else:
         server = url
     if ':' in server:
-        (server, port) = server.split(':')
+        (server, ports) = server.split(':')
```
Comment (dmjacobsen) [inline imagegw/shifter_imagegw/dockerv2.py]: it might be safer to limit the split to just one split, e.g.,
`(server,port) = server.split(':', 1)`

Comment (dmjacobsen) [inline imagegw/shifter_imagegw/dockerv2.py]: also, "ports" is pluralized there, which is probably a bug

Comment (scanon) [inline imagegw/shifter_imagegw/dockerv2.py]: The ports was a string version which gets cast to an int.  I renamed it to be a little more clear.  I also acted on your other suggestion.

Comment (dmjacobsen) [inline imagegw/shifter_imagegw/dockerv2.py]: looks good;  sorry i missed that was the portstr -- got so wrapped up in the new github review interface i missed the big picture =)

**Pull request merged**